src/util/getNodesOfKind.ts

```diff
@@ -0,0 +1,11 @@
+import { SourceFile, SyntaxKind, Node } from "ts-morph";
+
+export function getNodesOfKind(node: SourceFile, kind: SyntaxKind): Node[] {
+  const out: Node[] = [];
+  node.forEachDescendant((node) => {
```

Comment:
I have a slight preference for using immutable operations for things like these. Provided that `.descendants()` is available:

For example:

```ts
export function getNodesOfKind(node: SourceFile, kind: SyntaxKind): Node[] {
   return node.descendants().filter(node => node.getKind() === kind);
}
```